fe/fe-core/src/main/java/org/apache/doris/load/loadv2/LoadManager.java

```diff
@@ -399,13 +401,26 @@ public long getLoadJobNum(JobState jobState, EtlJobType jobType) {
      **/
     public void removeOldLoadJob() {
         long currentTimeMs = System.currentTimeMillis();
+        removeLoadJobIf(job -> job.isExpired(currentTimeMs), -1);
+    }
+
+    /**
+     * Remove completed jobs if total job num exceed Config.label_num_threshold
+     */
+    public void removeOverLimitLoadJob() {
+        if (idToLoadJob.size() <= Config.label_num_threshold) {
+            return;
+        }
+        removeLoadJobIf(LoadJob::isCompleted, Config.label_num_threshold);
+    }
 
+    private void removeLoadJobIf(Predicate<LoadJob> pred, int numLimit) {
         writeLock();
         try {
             Iterator<Map.Entry<Long, LoadJob>> iter = idToLoadJob.entrySet().iterator();
-            while (iter.hasNext()) {
+            while (iter.hasNext() && idToLoadJob.size() > numLimit) {
```

Comment:
do we need think about sorting by finishTime ? I think we keep the last recently label_num_threshold final job maybe better